packages/vertexai/src/requests/request.ts

```diff
@@ -140,24 +140,38 @@ export async function makeRequest(
     response = await fetch(request.url, request.fetchOptions);
     if (!response.ok) {
       let message = '';
+      let errorDetails;
       try {
         const json = await response.json();
         message = json.error.message;
         if (json.error.details) {
           message += ` ${JSON.stringify(json.error.details)}`;
+          errorDetails = json.error.details;
         }
       } catch (e) {
         // ignored
       }
-      throw new Error(`[${response.status} ${response.statusText}] ${message}`);
+      throw new VertexAIError(
+        VertexAIErrorCode.FETCH_ERROR,
+        `Error fetching from ${url}: [${response.status} ${response.statusText}] ${message}`,
+        response.status,
+        response.statusText,
+        errorDetails
+      );
     }
-  } catch (caughtError) {
-    const e = caughtError as Error;
-    const err = ERROR_FACTORY.create(VertexError.FETCH_ERROR, {
-      url: url.toString(),
-      message: e.message
-    });
-    err.stack = e.stack;
+  } catch (e) {
+    let err = e as Error;
+    if (
+      (e as FirebaseError).code !== VertexAIErrorCode.FETCH_ERROR &&
```

Comment:
This can be VertexAIError, right? Then you don't have to import FirebaseError.

Reply:
Fixed